httpx/_transports/jsfetch.py

```diff
@@ -0,0 +1,467 @@
+            else:
+                this_buffer = result_js.value.to_py()
+                yield this_buffer
+
+    async def aclose(self) -> None:
+        self._stream_js = None
+
+
+class AsyncJavascriptFetchTransport(AsyncBaseTransport):
+    def __init__(
+        self,
+        verify: ssl.SSLContext | str | bool = True,
+        cert: CertTypes | None = None,
+        trust_env: bool = True,
+        http1: bool = True,
+        http2: bool = False,
+        limits: Limits = DEFAULT_LIMITS,
+        proxy: ProxyTypes | None = None,
+        uds: str | None = None,
+        local_address: str | None = None,
+        retries: int = 0,
+        socket_options: typing.Iterable[SOCKET_OPTION] | None = None,
+    ) -> None:
+        pass
+
+    async def __aenter__(self: A) -> A:  # Use generics for subclass support.
+        return self
+
+    async def __aexit__(
+        self,
+        exc_type: type[BaseException] | None = None,
+        exc_value: BaseException | None = None,
+        traceback: TracebackType | None = None,
+    ) -> None:
+        pass
+
+    async def handle_async_request(
+        self,
+        request: Request,
+    ) -> Response:
+        assert isinstance(request.stream, AsyncByteStream)
+        body_data: bytes = b""
+        async for x in request.stream:
+            body_data += x
+        if len(body_data) == 0:
+            req_body = None
+        else:
+            req_body = body_data
+        conn_timeout = 0.0
+        read_timeout = 0.0
+        if "timeout" in request.extensions:
+            timeout_dict = request.extensions["timeout"]
+            if timeout_dict is not None:
+                if "connect" in timeout_dict:
+                    conn_timeout = timeout_dict["connect"]
+                if "read" in timeout_dict:
+                    read_timeout = timeout_dict["connect"]
+
+        abort_controller_js = js.AbortController.new()
+        headers = {
+            k: v for k, v in request.headers.items() if k not in HEADERS_TO_IGNORE
+        }
+        fetch_data = {
+            "headers": headers,
+            "body": pyodide.ffi.to_js(req_body),
+            "method": request.method,
+            "signal": abort_controller_js.signal,
+        }
+
+        fetcher_promise_js = js.fetch(
+            request.url,
+            pyodide.ffi.to_js(fetch_data, dict_converter=js.Object.fromEntries),
+        )
+        response_js = await _run_async_with_timeout(
+            fetcher_promise_js,
+            conn_timeout,
+            abort_controller_js,
+            ConnectTimeout,
+            ConnectError,
+        )
+
+        headers = {}
+        header_iter = response_js.headers.entries()
+        while True:
+            iter_value_js = header_iter.next()
+            if getattr(iter_value_js, "done", False):
+                break
+            else:
+                headers[str(iter_value_js.value[0])] = str(iter_value_js.value[1])
+        status_code = response_js.status
```

Comment:
We need to fix the content encoding here as well:
```py
        # fix content-encoding headers because the javascript fetch handles that
        headers["content-encoding"] = "identity"
```